src/modules/kvs-watch/kvs-watch.c

```diff
@@ -442,6 +457,23 @@ static int handle_initial_response (flux_t *h,
         }
 
         w->initial_rootseq = root_seq;
+
+        if (w->flags & FLUX_KVS_WATCH_INITIAL_SENTINEL
+            && w->initial_sentinel_sent == false) {
+            /* We want to send sentinel on the last load from
+             * load_range() call above.  We set an aux "flag" to note
+             * to other code to send the sentinel after this future is
+             * completed.  N.B. use the future as a random pointer to
+             * set for aux, we don't actually use the "flag" value.
+             */
+            flux_future_t *f = zlist_tail (w->loads);
+            if (flux_future_aux_set (f, "initial_sentinel", f, NULL) < 0) {
+                flux_log_error (h,
+                                "%s: failed to set initial_sentinel flag",
+                                __FUNCTION__);
+                goto error_respond;
+            }
```

Comment:
Instead of dropping a real pointer in that aux entry and having to explain it, maybe it would better to make it an obvious fake, like
```c
#include "ccan/ptrint/ptrint.h"
flux_future_aux_set (f, "initial_sentinel", int2ptr (1), NULL);
```